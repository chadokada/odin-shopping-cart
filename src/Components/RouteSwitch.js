import { BrowserRouter, Routes, Route } from "react-router-dom";
import Nav from './Nav';
import App from './App';
import Hats from './Hats'

const RouteSwitch = () => {
  return (
    
    <BrowserRouter>
      <Nav />
      <Routes>
        <Route path="/hats" element={<Hats />} />
      </Routes>
    </BrowserRouter>
  )
}

export default RouteSwitch;