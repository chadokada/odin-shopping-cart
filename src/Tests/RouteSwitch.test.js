/* eslint-disable testing-library/no-node-access */
import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { shallow } from 'enzyme';
import { MemoryRouter, Route, Routes } from 'react-router-dom';
import RouteSwitch from '../Components/RouteSwitch';

describe('Test route switch', () => {
  render(<RouteSwitch />);
  const shopNowButton = screen.getByRole('button');
  userEvent.click(shopNowButton);
  const fugg = screen.getByRole('button', {name: 'store-namee'})

  it('should render store name correctly', () => {
    const shopName = screen.getByTestId('store-name')
    expect(shopName.textContent).toBe("Butter and Sugar")
  });


})


