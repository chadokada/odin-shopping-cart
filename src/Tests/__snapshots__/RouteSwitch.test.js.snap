// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test route switch should match snapshot? 1`] = `<div />`;
